src/OrchardCore.Modules/Orchard.Autoroute/Indexes/AutoroutePartIndex.cs

```diff
@@ -18,7 +18,7 @@ public override void Describe(DescribeContext<ContentItem> context)
                 .Map(contentItem =>
                 {
                     var path = contentItem.As<AutoroutePart>()?.Path;
-                    if (!String.IsNullOrEmpty(path))
+                    if (!String.IsNullOrEmpty(path) && (contentItem.Published || contentItem.Latest))
```

Comment:
Not sure it's good. We should actually only have Published items here. But going where you might have already been I get that there is no obvious solution here.
So I suggest to add a bool Published in the index, so that we can at least filter on published items when we build the Route (I know it was not the case before your PR, it's wrong). You keep this filter but assign Published = true if the item is the published version. (Otherwise it's a draft).